File: 0806. Number of Lines To Write String/Solution.java
class Solution {
    public int[] numberOfLines(int[] widths, String S) {
        int lines = 1, width = 0;
        for (char ch : S.toCharArray()) {
            if (width + widths[ch - 'a'] > 100) {
                width = 0;
                lines++;
            }
            width += widths[ch - 'a'];
        }
        return new int[]{lines, width};
    }
}

File: 0806. Number of Lines To Write String/Solution.py
from typing import List


class Solution:
    def numberOfLines(self, widths: List[int], S: str) -> List[int]:
        lines, width = 1, 0
        for ch in S:
            if width + widths[ord(ch) - ord('a')] > 100:
                width = 0
                lines += 1
            width += widths[ord(ch) - ord('a')]
        return [lines, width]
